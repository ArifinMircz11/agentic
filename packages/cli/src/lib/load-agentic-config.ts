import {
  type AgenticProjectConfig,
  validateAgenticProjectConfig
} from '@agentic/platform-schemas'
import { loadConfig } from 'unconfig'

export async function loadAgenticConfig({
  cwd
}: {
  cwd?: string
} = {}): Promise<AgenticProjectConfig> {
  const { config } = await loadConfig({
    cwd,
    sources: [
      {
        files: 'agentic.config',
        extensions: ['ts', 'mts', 'cts', 'js', 'mjs', 'cjs', 'json']
      }
    ]
  })

  return validateAgenticProjectConfig(config)
}
