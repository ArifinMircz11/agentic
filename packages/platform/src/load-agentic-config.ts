import type { AgenticProjectConfig } from '@agentic/platform-types'
import { loadConfig } from 'unconfig'

import { validateAgenticProjectConfig } from './validate-agentic-project-config'

export async function loadAgenticConfig({
  cwd
}: {
  cwd?: string
} = {}): Promise<AgenticProjectConfig> {
  const { config } = await loadConfig({
    cwd,
    sources: [
      {
        files: 'agentic.config',
        extensions: ['ts', 'mts', 'cts', 'js', 'mjs', 'cjs', 'json']
      }
    ]
  })

  return validateAgenticProjectConfig(config)
}
