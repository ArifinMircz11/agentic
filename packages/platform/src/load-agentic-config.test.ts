import path from 'node:path'
import { fileURLToPath } from 'node:url'

import { describe, expect, test } from 'vitest'

import { loadAgenticConfig } from './load-agentic-config'

const fixtures = [
  'basic-raw-free-ts',
  'basic-raw-free-json',
  'pricing-freemium',
  'pricing-pay-as-you-go',
  'pricing-3-plans',
  'pricing-monthly-annual',
  'pricing-custom-0',
  'basic-openapi',
  'basic-mcp',
  'everything-openapi',
  'metadata-0',
  'metadata-1',
  'metadata-2'
]

const invalidFixtures = [
  'pricing-base-inconsistent',
  'pricing-custom-inconsistent',
  'pricing-empty-0',
  'pricing-empty-1',
  'pricing-empty-2',
  'pricing-duplicate-0',
  'pricing-duplicate-1',
  'invalid-origin-url-0',
  'invalid-origin-url-1',
  'invalid-origin-url-2',
  'invalid-origin-url-3',
  'invalid-name-0',
  'invalid-name-1',
  'invalid-name-2',
  'invalid-slug-0',
  'invalid-slug-1',
  'invalid-slug-2',
  'invalid-slug-3',
  'invalid-slug-4',
  'invalid-metadata-0',
  'invalid-metadata-1'
]

const fixturesDir = path.join(
  fileURLToPath(import.meta.url),
  '..',
  '..',
  '..',
  '..',
  'fixtures'
)

const validFixturesDir = path.join(fixturesDir, 'valid')
const invalidFixturesDir = path.join(fixturesDir, 'invalid')

describe('loadAgenticConfig', () => {
  for (const fixture of fixtures) {
    test(
      `${fixture}`,
      {
        timeout: 60_000
      },
      async () => {
        const fixtureDir = path.join(validFixturesDir, fixture)

        const config = await loadAgenticConfig({ cwd: fixtureDir })
        expect(config).toMatchSnapshot()
      }
    )
  }

  for (const fixture of invalidFixtures) {
    test(
      `invalid: ${fixture}`,
      {
        timeout: 60_000
      },
      async () => {
        const fixtureDir = path.join(invalidFixturesDir, fixture)

        await expect(
          loadAgenticConfig({ cwd: fixtureDir })
        ).rejects.toThrowErrorMatchingSnapshot()
      }
    )
  }
})
