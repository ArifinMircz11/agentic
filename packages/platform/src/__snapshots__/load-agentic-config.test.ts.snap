// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`loadAgenticConfig > basic-mcp 1`] = `
{
  "defaultRateLimit": {
    "enabled": true,
    "interval": 60,
    "limit": 1000,
    "mode": "approximate",
  },
  "name": "test-basic-mcp",
  "origin": {
    "location": "external",
    "type": "mcp",
    "url": "https://agentic-basic-mcp-test.onrender.com/mcp",
  },
  "pricingIntervals": [
    "month",
  ],
  "pricingPlans": [
    {
      "lineItems": [
        {
          "amount": 0,
          "slug": "base",
          "usageType": "licensed",
        },
      ],
      "name": "Free",
      "rateLimit": {
        "enabled": true,
        "interval": 60,
        "limit": 1000,
        "mode": "approximate",
      },
      "slug": "free",
    },
  ],
  "toolConfigs": [],
  "version": undefined,
}
`;

exports[`loadAgenticConfig > basic-openapi 1`] = `
{
  "defaultRateLimit": {
    "enabled": true,
    "interval": 60,
    "limit": 1000,
    "mode": "approximate",
  },
  "name": "test-basic-openapi",
  "origin": {
    "location": "external",
    "spec": "{"openapi":"3.1.0","info":{"title":"JSONPlaceholder","version":"1.0.0"},"paths":{"/posts":{"get":{"summary":"Get posts","operationId":"getPosts","responses":{"200":{"description":"Success","content":{"application/json":{"schema":{"type":"array","items":{"$ref":"#/components/schemas/Post"}}}}}}},"post":{"summary":"Create post","operationId":"createPost","requestBody":{"required":true,"content":{"application/json":{"schema":{"type":"object","required":["userId","title","body"],"properties":{"userId":{"type":"integer"},"title":{"type":"string"},"body":{"type":"string"}}}}}},"responses":{"200":{"description":"Success","content":{"application/json":{"schema":{"$ref":"#/components/schemas/Post"}}}}}}},"/posts/{postId}":{"get":{"summary":"Get post","operationId":"getPost","parameters":[{"required":true,"schema":{"type":"integer"},"name":"postId","in":"path"}],"responses":{"200":{"description":"Success","content":{"application/json":{"schema":{"$ref":"#/components/schemas/Post"}}}}}}}},"components":{"schemas":{"Post":{"type":"object","required":["id","userId","title","body"],"properties":{"id":{"type":"integer"},"userId":{"type":"integer"},"title":{"type":"string"},"body":{"type":"string"}}}}}}",
    "type": "openapi",
    "url": "https://jsonplaceholder.typicode.com",
  },
  "pricingIntervals": [
    "month",
  ],
  "pricingPlans": [
    {
      "lineItems": [
        {
          "amount": 0,
          "slug": "base",
          "usageType": "licensed",
        },
      ],
      "name": "Free",
      "rateLimit": {
        "enabled": true,
        "interval": 60,
        "limit": 1000,
        "mode": "approximate",
      },
      "slug": "free",
    },
  ],
  "toolConfigs": [
    {
      "name": "get_posts",
      "pure": true,
    },
    {
      "name": "get_post",
      "pure": true,
    },
  ],
  "version": undefined,
}
`;

exports[`loadAgenticConfig > basic-raw-free-json 1`] = `
{
  "defaultRateLimit": {
    "enabled": true,
    "interval": 60,
    "limit": 1000,
    "mode": "approximate",
  },
  "name": "test-basic-raw-free-json",
  "origin": {
    "location": "external",
    "type": "raw",
    "url": "https://jsonplaceholder.typicode.com",
  },
  "pricingIntervals": [
    "month",
  ],
  "pricingPlans": [
    {
      "lineItems": [
        {
          "amount": 0,
          "slug": "base",
          "usageType": "licensed",
        },
      ],
      "name": "Free",
      "rateLimit": {
        "enabled": true,
        "interval": 60,
        "limit": 1000,
        "mode": "approximate",
      },
      "slug": "free",
    },
  ],
  "toolConfigs": [],
  "version": undefined,
}
`;

exports[`loadAgenticConfig > basic-raw-free-ts 1`] = `
{
  "defaultRateLimit": {
    "enabled": true,
    "interval": 60,
    "limit": 1000,
    "mode": "approximate",
  },
  "name": "test-basic-raw-free-ts",
  "origin": {
    "location": "external",
    "type": "raw",
    "url": "https://jsonplaceholder.typicode.com",
  },
  "pricingIntervals": [
    "month",
  ],
  "pricingPlans": [
    {
      "lineItems": [
        {
          "amount": 0,
          "slug": "base",
          "usageType": "licensed",
        },
      ],
      "name": "Free",
      "rateLimit": {
        "enabled": true,
        "interval": 60,
        "limit": 1000,
        "mode": "approximate",
      },
      "slug": "free",
    },
  ],
  "toolConfigs": [],
  "version": undefined,
}
`;

exports[`loadAgenticConfig > everything-openapi 1`] = `
{
  "defaultRateLimit": {
    "enabled": true,
    "interval": 60,
    "limit": 1000,
    "mode": "approximate",
  },
  "name": "test-everything-openapi",
  "origin": {
    "location": "external",
    "spec": "{"openapi":"3.1.0","info":{"title":"OpenAPI server everything","description":"OpenAPI kitchen sink server meant for testing Agentic's origin OpenAPI adapter and ToolConfig features.","version":"0.1.0"},"components":{"schemas":{"User":{"type":"object","properties":{"id":{"type":"string"},"name":{"type":"string"},"email":{"type":"string"}},"required":["id","name","email"]}},"parameters":{}},"paths":{"/health":{"get":{"description":"Check if the server is healthy","operationId":"healthCheck","responses":{"200":{"description":"OK","content":{"application/json":{"schema":{"type":"object","properties":{"status":{"type":"string"}},"required":["status"]}}}}}}},"/users/{userId}":{"get":{"description":"Gets a user","tags":["users"],"operationId":"getUser","parameters":[{"schema":{"type":"string"},"required":true,"description":"User ID","name":"userId","in":"path"}],"responses":{"200":{"description":"A user object","content":{"application/json":{"schema":{"$ref":"#/components/schemas/User"}}}}}}},"/disabled-tool":{"get":{"description":"Disabled tool","operationId":"disabledTool","responses":{"200":{"description":"OK","content":{"application/json":{"schema":{"type":"object","properties":{"status":{"type":"string"}},"required":["status"]}}}}}}},"/disabled-for-free-plan-tool":{"get":{"description":"Disabled for free plan tool","operationId":"disabledForFreePlanTool","responses":{"200":{"description":"OK","content":{"application/json":{"schema":{"type":"object","properties":{"status":{"type":"string"}},"required":["status"]}}}}}}},"/echo":{"post":{"description":"Echoes the request body","operationId":"echo","requestBody":{"content":{"application/json":{"schema":{"type":"object","properties":{}}}}},"responses":{"200":{"description":"Echoed request body","content":{"application/json":{"schema":{"type":"object","properties":{}}}}}}}},"/echo-headers":{"get":{"description":"Echoes the request headers","operationId":"echoHeaders","responses":{"200":{"description":"Echoed request headers","content":{"application/json":{"schema":{"type":"object","properties":{}}}}}}}},"/pure":{"post":{"description":"Pure tool","operationId":"pure","requestBody":{"content":{"application/json":{"schema":{"type":"object","properties":{}}}}},"responses":{"200":{"description":"Echoed request body","content":{"application/json":{"schema":{"type":"object","properties":{}}}}}}}},"/unpure-marked-pure":{"post":{"description":"Unpure tool marked pure","operationId":"unpure_marked_pure","requestBody":{"content":{"application/json":{"schema":{"type":"object","properties":{}}}}},"responses":{"200":{"description":"Echoed request body with current timestamp to not be pure","content":{"application/json":{"schema":{"type":"object","properties":{"now":{"type":"number"}},"required":["now"]}}}}}}},"/custom-cache-control-tool":{"post":{"description":"Custom cache control tool","operationId":"customCacheControlTool","requestBody":{"content":{"application/json":{"schema":{"type":"object","properties":{}}}}},"responses":{"200":{"description":"Echoed request body","content":{"application/json":{"schema":{"type":"object","properties":{}}}}}}}},"/custom-rate-limit-approximate-tool":{"post":{"description":"Custom rate limit tool (approximate mode)","operationId":"customRateLimitApproximateTool","requestBody":{"content":{"application/json":{"schema":{"type":"object","properties":{}}}}},"responses":{"200":{"description":"Echoed request body","content":{"application/json":{"schema":{"type":"object","properties":{}}}}}}}},"/no-store-cache-control-tool":{"post":{"description":"No store cache control tool","operationId":"noStoreCacheControlTool","requestBody":{"content":{"application/json":{"schema":{"type":"object","properties":{}}}}},"responses":{"200":{"description":"Echoed request body","content":{"application/json":{"schema":{"type":"object","properties":{}}}}}}}},"/no-cache-cache-control-tool":{"post":{"description":"No cache cache control tool","operationId":"noCacheCacheControlTool","requestBody":{"content":{"application/json":{"schema":{"type":"object","properties":{}}}}},"responses":{"200":{"description":"Echoed request body","content":{"application/json":{"schema":{"type":"object","properties":{}}}}}}}},"/custom-rate-limit-tool":{"post":{"description":"Custom rate limit tool (strict mode)","operationId":"customRateLimitTool","requestBody":{"content":{"application/json":{"schema":{"type":"object","properties":{}}}}},"responses":{"200":{"description":"Echoed request body","content":{"application/json":{"schema":{"type":"object","properties":{}}}}}}}},"/disabled-rate-limit-tool":{"post":{"description":"Disabled rate limit tool","operationId":"disabledRateLimitTool","requestBody":{"content":{"application/json":{"schema":{"type":"object","properties":{}}}}},"responses":{"200":{"description":"Echoed request body","content":{"application/json":{"schema":{"type":"object","properties":{}}}}}}}},"/strict-additional-properties":{"post":{"description":"Echoes the request body only allowing a single \\"foo\\" field.","operationId":"strictAdditionalProperties","requestBody":{"content":{"application/json":{"schema":{"type":"object","properties":{"foo":{"type":"string"}},"required":["foo"]}}}},"responses":{"200":{"description":"Echoed request body","content":{"application/json":{"schema":{"type":"object","properties":{"foo":{"type":"string"}},"required":["foo"]}}}}}}}},"webhooks":{}}",
    "type": "openapi",
    "url": "https://agentic-platform-fixtures-everything.onrender.com",
  },
  "pricingIntervals": [
    "month",
  ],
  "pricingPlans": [
    {
      "lineItems": [
        {
          "amount": 0,
          "slug": "base",
          "usageType": "licensed",
        },
        {
          "billingScheme": "per_unit",
          "slug": "requests",
          "unitAmount": 0,
          "usageType": "metered",
        },
      ],
      "name": "Free",
      "slug": "free",
    },
    {
      "interval": "month",
      "lineItems": [
        {
          "amount": 999,
          "slug": "base",
          "usageType": "licensed",
        },
        {
          "billingScheme": "tiered",
          "slug": "requests",
          "tiers": [
            {
              "unitAmount": 0,
              "upTo": 1000,
            },
            {
              "unitAmount": 0.053,
              "upTo": "inf",
            },
          ],
          "tiersMode": "volume",
          "usageType": "metered",
        },
      ],
      "name": "Starter",
      "slug": "starter",
    },
    {
      "interval": "month",
      "lineItems": [
        {
          "amount": 2999,
          "slug": "base",
          "usageType": "licensed",
        },
        {
          "billingScheme": "tiered",
          "slug": "requests",
          "tiers": [
            {
              "unitAmount": 0,
              "upTo": 10000,
            },
            {
              "unitAmount": 0.049,
              "upTo": "inf",
            },
          ],
          "tiersMode": "volume",
          "usageType": "metered",
        },
      ],
      "name": "Pro",
      "slug": "pro",
    },
  ],
  "toolConfigs": [
    {
      "enabled": true,
      "name": "get_user",
      "pricingPlanOverridesMap": {
        "free": {
          "enabled": true,
          "reportUsage": true,
        },
      },
      "pure": true,
      "rateLimit": {
        "enabled": false,
      },
      "reportUsage": true,
    },
    {
      "enabled": false,
      "name": "disabled_tool",
    },
    {
      "name": "disabled_for_free_plan_tool",
      "pricingPlanOverridesMap": {
        "free": {
          "enabled": false,
        },
      },
    },
    {
      "name": "pure",
      "pure": true,
    },
    {
      "name": "unpure_marked_pure",
      "pure": true,
    },
    {
      "cacheControl": "public, max-age=7200, s-maxage=7200, stale-while-revalidate=3600",
      "name": "custom_cache_control_tool",
    },
    {
      "cacheControl": "no-cache",
      "name": "no_cache_cache_control_tool",
    },
    {
      "cacheControl": "no-store",
      "name": "no_store_cache_control_tool",
    },
    {
      "name": "custom_rate_limit_tool",
      "rateLimit": {
        "enabled": true,
        "interval": 30,
        "limit": 2,
        "mode": "strict",
      },
    },
    {
      "name": "custom_rate_limit_approximate_tool",
      "rateLimit": {
        "enabled": true,
        "interval": 30,
        "limit": 2,
        "mode": "approximate",
      },
    },
    {
      "name": "disabled_rate_limit_tool",
      "rateLimit": {
        "enabled": false,
      },
    },
    {
      "inputSchemaAdditionalProperties": false,
      "name": "strict_additional_properties",
      "outputSchemaAdditionalProperties": false,
    },
  ],
  "version": undefined,
}
`;

exports[`loadAgenticConfig > invalid: invalid-name-0 1`] = `[Error: Invalid project name "Test Invalid Name 0". Must be ascii-only lower kebab-case with no spaces between 1 and 256 characters. For example: "my-project" or "linkedin-resolver-23"]`;

exports[`loadAgenticConfig > invalid: invalid-name-1 1`] = `[Error: Invalid project name "Test-Invalid-Name-1". Must be ascii-only lower kebab-case with no spaces between 1 and 256 characters. For example: "my-project" or "linkedin-resolver-23"]`;

exports[`loadAgenticConfig > invalid: invalid-name-2 1`] = `[Error: Invalid project name "test_invalid_name_2". Must be ascii-only lower kebab-case with no spaces between 1 and 256 characters. For example: "my-project" or "linkedin-resolver-23"]`;

exports[`loadAgenticConfig > invalid: invalid-name-3 1`] = `[ZodValidationError: Validation error: Required at "name"]`;

exports[`loadAgenticConfig > invalid: invalid-name-4 1`] = `[Error: Invalid project name "@foo/bar". Must be ascii-only lower kebab-case with no spaces between 1 and 256 characters. For example: "my-project" or "linkedin-resolver-23"]`;

exports[`loadAgenticConfig > invalid: invalid-origin-url-0 1`] = `[Error: Invalid originUrl: must be a valid https URL]`;

exports[`loadAgenticConfig > invalid: invalid-origin-url-1 1`] = `[ZodValidationError: Validation error: Invalid url at "origin.url"]`;

exports[`loadAgenticConfig > invalid: invalid-origin-url-2 1`] = `[ZodValidationError: Validation error: Required at "origin"]`;

exports[`loadAgenticConfig > invalid: invalid-origin-url-3 1`] = `[ZodValidationError: Validation error: Invalid url at "origin.url"]`;

exports[`loadAgenticConfig > invalid: pricing-base-inconsistent 1`] = `[ZodValidationError: Validation error: Invalid PricingPlanLineItem "base": reserved "base" LineItems must have "licensed" usage type. at "pricingPlans[1].lineItems[0]"]`;

exports[`loadAgenticConfig > invalid: pricing-custom-inconsistent 1`] = `[Error: Invalid pricingPlans: all PricingPlans which contain the same LineItems (by slug "custom-test") must have the same usage type ("licensed" or "metered").]`;

exports[`loadAgenticConfig > invalid: pricing-duplicate-0 1`] = `[Error: Invalid pricingPlans: duplicate PricingPlan slugs. All PricingPlan slugs must be unique (e.g. "free", "starter-monthly", "pro-annual", etc).]`;

exports[`loadAgenticConfig > invalid: pricing-duplicate-1 1`] = `[Error: Invalid pricingPlan "basic": duplicate line-item slugs]`;

exports[`loadAgenticConfig > invalid: pricing-empty-0 1`] = `[ZodValidationError: Validation error: Must contain at least one pricing interval at "pricingIntervals"]`;

exports[`loadAgenticConfig > invalid: pricing-empty-1 1`] = `[ZodValidationError: Validation error: Array must contain at least 1 element(s) at "pricingPlans[0].lineItems"]`;

exports[`loadAgenticConfig > invalid: pricing-empty-2 1`] = `[ZodValidationError: Validation error: Must contain at least one PricingPlan at "pricingPlans"]`;

exports[`loadAgenticConfig > pricing-3-plans 1`] = `
{
  "defaultRateLimit": {
    "enabled": true,
    "interval": 60,
    "limit": 1000,
    "mode": "approximate",
  },
  "name": "test-pricing-3-plans",
  "origin": {
    "location": "external",
    "type": "raw",
    "url": "https://httpbin.org",
  },
  "pricingIntervals": [
    "month",
  ],
  "pricingPlans": [
    {
      "features": [
        "Unlimited images **with watermark**",
        "No backend generation",
        "No custom templates\`",
      ],
      "lineItems": [
        {
          "amount": 0,
          "slug": "base",
          "usageType": "licensed",
        },
      ],
      "name": "Free",
      "slug": "free",
    },
    {
      "features": [
        "Unlimited images **without watermark**",
        "Unlimited backend generation",
        "No custom templates",
      ],
      "interval": "month",
      "lineItems": [
        {
          "amount": 999,
          "slug": "base",
          "usageType": "licensed",
        },
      ],
      "name": "Starter",
      "slug": "starter",
    },
    {
      "features": [
        "Unlimited images **without watermark**",
        "Unlimited backend generation",
        "Unlimited custom templates",
      ],
      "interval": "month",
      "lineItems": [
        {
          "amount": 2999,
          "slug": "base",
          "usageType": "licensed",
        },
      ],
      "name": "Pro",
      "slug": "pro",
    },
  ],
  "toolConfigs": [],
  "version": undefined,
}
`;

exports[`loadAgenticConfig > pricing-custom-0 1`] = `
{
  "defaultRateLimit": {
    "enabled": true,
    "interval": 60,
    "limit": 1000,
    "mode": "approximate",
  },
  "name": "test-pricing-custom-0",
  "origin": {
    "location": "external",
    "type": "raw",
    "url": "https://httpbin.org",
  },
  "pricingIntervals": [
    "month",
    "year",
  ],
  "pricingPlans": [
    {
      "lineItems": [
        {
          "amount": 0,
          "slug": "base",
          "usageType": "licensed",
        },
      ],
      "name": "Free",
      "slug": "free",
    },
    {
      "interval": "month",
      "lineItems": [
        {
          "amount": 500,
          "slug": "base",
          "usageType": "licensed",
        },
        {
          "billingScheme": "per_unit",
          "slug": "custom-test",
          "unitAmount": 100,
          "usageType": "metered",
        },
      ],
      "name": "Basic Monthly",
      "rateLimit": {
        "enabled": true,
        "interval": 2592000,
        "limit": 1000,
        "mode": "approximate",
      },
      "slug": "basic-monthly",
    },
    {
      "interval": "year",
      "lineItems": [
        {
          "amount": 4800,
          "slug": "base",
          "usageType": "licensed",
        },
        {
          "billingScheme": "per_unit",
          "slug": "custom-test",
          "unitAmount": 80,
          "usageType": "metered",
        },
      ],
      "name": "Basic Annual",
      "rateLimit": {
        "enabled": true,
        "interval": 2592000,
        "limit": 1500,
        "mode": "approximate",
      },
      "slug": "basic-annual",
    },
  ],
  "toolConfigs": [],
  "version": undefined,
}
`;

exports[`loadAgenticConfig > pricing-freemium 1`] = `
{
  "defaultRateLimit": {
    "enabled": true,
    "interval": 60,
    "limit": 1000,
    "mode": "approximate",
  },
  "name": "test-pricing-freemium",
  "origin": {
    "location": "external",
    "type": "raw",
    "url": "https://httpbin.org",
  },
  "pricingIntervals": [
    "month",
  ],
  "pricingPlans": [
    {
      "lineItems": [
        {
          "amount": 0,
          "slug": "base",
          "usageType": "licensed",
        },
      ],
      "name": "Free",
      "rateLimit": {
        "enabled": true,
        "interval": 60,
        "limit": 1000,
        "mode": "approximate",
      },
      "slug": "free",
    },
    {
      "interval": "month",
      "lineItems": [
        {
          "amount": 499,
          "slug": "base",
          "usageType": "licensed",
        },
      ],
      "name": "Basic",
      "slug": "basic",
      "trialPeriodDays": 7,
    },
  ],
  "toolConfigs": [],
  "version": undefined,
}
`;

exports[`loadAgenticConfig > pricing-monthly-annual 1`] = `
{
  "defaultRateLimit": {
    "enabled": true,
    "interval": 60,
    "limit": 1000,
    "mode": "approximate",
  },
  "name": "test-pricing-monthly-annual",
  "origin": {
    "location": "external",
    "type": "raw",
    "url": "https://httpbin.org",
  },
  "pricingIntervals": [
    "month",
    "year",
  ],
  "pricingPlans": [
    {
      "lineItems": [
        {
          "amount": 0,
          "slug": "custom-base",
          "usageType": "licensed",
        },
      ],
      "name": "Free",
      "slug": "free",
    },
    {
      "interval": "month",
      "lineItems": [
        {
          "amount": 100,
          "slug": "custom-base",
          "usageType": "licensed",
        },
      ],
      "name": "Basic Monthly",
      "slug": "basic-monthly",
    },
    {
      "interval": "year",
      "lineItems": [
        {
          "amount": 70,
          "slug": "custom-base",
          "usageType": "licensed",
        },
      ],
      "name": "Basic Annual",
      "slug": "basic-annual",
    },
  ],
  "toolConfigs": [],
  "version": undefined,
}
`;

exports[`loadAgenticConfig > pricing-pay-as-you-go 1`] = `
{
  "defaultRateLimit": {
    "enabled": true,
    "interval": 60,
    "limit": 1000,
    "mode": "approximate",
  },
  "name": "test-pricing-pay-as-you-go",
  "origin": {
    "location": "external",
    "type": "raw",
    "url": "https://httpbin.org",
  },
  "pricingIntervals": [
    "month",
  ],
  "pricingPlans": [
    {
      "lineItems": [
        {
          "amount": 0,
          "slug": "base",
          "usageType": "licensed",
        },
        {
          "billingScheme": "per_unit",
          "slug": "requests",
          "unitAmount": 0,
          "usageType": "metered",
        },
      ],
      "name": "Free",
      "rateLimit": {
        "enabled": true,
        "interval": 86400,
        "limit": 20,
        "mode": "approximate",
      },
      "slug": "free",
    },
    {
      "interval": "month",
      "lineItems": [
        {
          "billingScheme": "tiered",
          "slug": "requests",
          "tiers": [
            {
              "unitAmount": 0.467,
              "upTo": 999,
            },
            {
              "unitAmount": 0.053,
              "upTo": "inf",
            },
          ],
          "tiersMode": "volume",
          "usageType": "metered",
        },
      ],
      "name": "Pay-As-You-Go",
      "rateLimit": {
        "enabled": true,
        "interval": 86400,
        "limit": 1000,
        "mode": "approximate",
      },
      "slug": "pay-as-you-go",
    },
  ],
  "toolConfigs": [],
  "version": undefined,
}
`;
